--- a/package/network/services/dnsmasq/files/dnsmasq.init
+++ b/package/network/services/dnsmasq/files/dnsmasq.init
@@ -1173,6 +1173,20 @@ dnsmasq_start()
 	esac
 
 	procd_close_instance
+
+	config_get_bool dns_redirect "$cfg" dns_redirect 0
+	config_get dns_port "$cfg" port 53
+	if [ "$dns_redirect" = 1 ]; then
+		cat <<-EOF > /tmp/dnsmasq-dns-redirect.nft
+			table inet dnsmasq {
+				chain prerouting {
+					type nat hook prerouting priority -105; policy accept;
+					meta nfproto {ipv4, ipv6} udp dport 53 counter redirect to :$dns_port comment "DNSMASQ";
+				}
+			}
+		EOF
+		nft -f /tmp/dnsmasq-dns-redirect.nft
+	fi
 }
 
 dnsmasq_stop()
@@ -1190,6 +1204,16 @@ dnsmasq_stop()
 	rm -f ${BASEDHCPSTAMPFILE}.${cfg}.*.dhcp
 }
 
+nftables_clear()
+{
+	nft --check list table inet dnsmasq > /dev/null 2>&1
+
+	if [ $? -eq "0" ]; then
+		echo "Remove dnsmasq nat table"
+		nft delete table inet dnsmasq
+	fi
+}
+
 add_interface_trigger()
 {
 	local interface ignore
@@ -1260,6 +1284,7 @@ start_service() {
 }
 
 reload_service() {
+	nftables_clear
 	rc_procd start_service "$@"
 	procd_send_signal dnsmasq "$@"
 }
@@ -1286,4 +1311,5 @@ stop_service() {
 	else
 		config_foreach dnsmasq_stop dnsmasq
 	fi
+	nftables_clear
 }
